File: C++/Medium/RuntimeError/maximum-ice-cream-bars_3.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        
        int n = costs.size();
        vector<int> count(100001, 0);
        for (int i=0; i<n; i++) {
            count[costs[i]]++;
        }
        int ans = 0;
        for (int i=1; i<=100000; i++) {
            if (count[i] == 0) continue;
            if (coins >= i * count[i]) {
                ans += count[i];
                coins -= i * count[i];
            } else {
                ans += coins / i;
                break;
            }
        }
        return ans;
    }
};

File: Java/Medium/Accepted/maximum-ice-cream-bars_3.java
class Solution {
    public int maxIceCream(int[] costs, int coins) {
        
        int[] count = new int[100001];
        for (int cost : costs) {
            count[cost]++;
        }
        int ans = 0;
        for (int i = 1; i < count.length; i++) {
            int c = Math.min(count[i], coins / i);
            ans += c;
            coins -= c * i;
        }
        return ans;
    }
}